### DISCUSSION TITLE
Can i get all the values sent in register?

### DISCUSSION THREAD

**Discussion by tkatariya** [react-hook-form/react-hook-form#1760] [Q&A]
I have a parent component which is registered to RHF, now i want to get value inside my child component that if i have passed required as true or false. With this value i want to show/hide some text in child component

**Comment by bluebill1049** [comment]
you want to use `watch` API?

**Reply by tkatariya** [reply]
i can, but will it give me if field is required?

**Reply by bluebill1049** [reply]
do you have a codesandbox?  i am not clear what you want.

**Reply by tkatariya** [reply]
https://codesandbox.io/s/react-hook-form-useform-template-uxqgl?file=/src/index.tsx

If you see i wanted to remove sublabel property from customtextfield and i wanted to show required(*) in customTextfield if we pass required:true in input ref.

I can do it other way too, but if we have handy solution that would be helpful

**Reply by bluebill1049** [reply]
I I still don't understand your question, your title is getting all the values sent in the register then I see your example with `forwardRef` with `regsiter`.

but with your CSB, you are using  `forwardRef`with `customInputRef`, you should only use `ref` because it's already forwarding the ref.
